zerver/tests/test_signup.py

```diff
@@ -1428,16 +1433,20 @@ def test_confirmation_key_of_wrong_type(self) -> None:
         self.assertEqual(cm.exception.error_type, ConfirmationKeyException.DOES_NOT_EXIST)
 
         # Verify that using the wrong type doesn't work in the main confirm code path
-        email_change_url = create_confirmation_link(user, 'host', Confirmation.EMAIL_CHANGE)
+        email_change_url = create_confirmation_link(prereg_user, 'host', Confirmation.EMAIL_CHANGE)
         email_change_key = email_change_url.split('/')[-1]
         url = '/accounts/do_confirm/' + email_change_key
         result = self.client_get(url)
         self.assert_in_success_response(["Whoops. We couldn't find your "
                                          "confirmation link in the system."], result)
 
     def test_confirmation_expired(self) -> None:
-        user = self.example_user('hamlet')
-        url = create_confirmation_link(user, 'host', Confirmation.USER_REGISTRATION)
+        email = self.nonreg_email("alice")
+        realm = get_realm('zulip')
+        inviter = UserProfile.objects.first()
+        prereg_user = PreregistrationUser.objects.create(
+            email=email, referred_by=inviter, realm=realm)
+        url = create_confirmation_link(prereg_user, 'host', Confirmation.USER_REGISTRATION)
```

Comment:
Can we have these test fixes in a separate, preliminary commit?

Reply:
I have already modified the commit history, will I still be able to add them as a part of a separate commit?

Reply:
Sure, you can rebase again :slightly_smiling_face: